### PR TITLE
Add new root CA to the signal test

### PR DESCRIPTION
**Pull request by hellais**
All measurements collected since 2022-10-19 with previous versions of OONI Probe will wrongly report sfu.voip.signal.org as blocked as it switched to using a different root CA

This fixes: https://github.com/ooni/probe/issues/2344

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/ooni/probe-cli/blob/master/CONTRIBUTING.md)
- [x] reference issue for this pull request: https://github.com/ooni/probe/issues/2344
- [x] if you changed anything related how experiments work and you need to reflect these changes in the ooni/spec repository, please link to the related ooni/spec pull request: https://github.com/ooni/spec/pull/265

### CODE REVIEW

File: internal/engine/experiment/signal/signal.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 
 const (
 	testName    = "signal"
-	testVersion = "0.2.0"
+	testVersion = "0.2.1"
```
Comment (bassosimone) [inline internal/engine/experiment/signal/signal.go:18]: You should also update the corresponding test in `signal_test.go`, which fails like this:

```
signal_test.go:21: unexpected version
```

File: internal/engine/experiment/signal/signal.go [hunk 2]
```diff
@@ -40,6 +40,34 @@ iOheyv7UzJOefb2pLOc9qsuvI4fnaESh9bhzln+LXxtCrRPGhkxA1IMIo3J/s2WF
...
+aelPHRa5xNPVfkrN8xw4fv/8TLE9GSjKlrNJm1EoTZL5CYWQU+qe4CuKfAJU6h8l
+xIkcik61aCeNLQoaI1L3V8tPXmmqMWpsnZmFg6YLGeMTLs4skdFqgLOnx9EF2jgO
+7EAJ9HcrgSPirQeuDJKhamaLtQiqIQR8L3H4YG1FDiuOeto6f1LRCIqjH1Mye1BM
+33Qg/VilLQIWp8+C4GJZ0+LO1cfatNh8tkDbrwMzUeA1nLEZHMlgXE05z00euNlQ
+0+evTmJzWRKJHugPnA3vvdzy4lbYvYWaXs8pACrVpESui8I+v6jdH814lOxpDwNH
+bPrxfOxhIxfFiVttCl3AQZBLJM6M0ty6/Q7bYsdNT23jKMl0AmDhj9qn/7dzYcVi
+vI0XKaaJl4ov3IDbuMe0oZWhoLwzPuWxxkWDjTb8ngDnWZT1o5dAR9fltr38m42N
+uA/SkxghiAMmvkC8nhEJ7yT2hme+rozPZSp1SSEDViDkA4KnnQpMcNiotCQpNOe7
+YfA9uSnjHjZloRTPUgtkKQ3u8ZZprFQlS2jDE18BRGdh24V5OsCbMvFPtrEsjG4H
+5xvkiIV0FpbMk4Gj8I4Hbw==
```
Comment (bassosimone) [inline internal/engine/experiment/signal/signal.go:70]: I see this failure in tests, which probably is related to adding a new CA:

```
--- FAIL: TestBadSignalCA (0.04s)
    signal_test.go:115: not the error we expected
FAIL
```

We should fix this failure before merging!

File: internal/engine/experiment/signal/signal.go (hunk 1 above)
Comment (hellais) [reply internal/engine/experiment/signal/signal.go:18]: Done!

File: internal/engine/experiment/signal/signal.go (hunk 2 above)
Comment (hellais) [reply internal/engine/experiment/signal/signal.go:70]: Ah good point. I missed the fact that this test can be configured with a custom CA supplied via a config option.

Based on this I refactored a bit the code to work a bit better for this case.

The tests are now fixed.

Comment (bassosimone) [reply internal/engine/experiment/signal/signal.go:70]: 🔝 💯 🙏 🥳

File: internal/engine/experiment/signal/signal.go (hunk 1 above)
Comment (bassosimone) [reply internal/engine/experiment/signal/signal.go:18]: 🙏

**Review by bassosimone — APPROVED** [review]
Thanks, I'll merge!

**Pull request merged**